pkg/machinepool/helper.go

```diff
@@ -0,0 +1,267 @@
+package machinepool
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	ec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	commonUtils "github.com/openshift-online/ocm-common/pkg/utils"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/interactive"
+	interactiveSgs "github.com/openshift/rosa/pkg/interactive/securitygroups"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+func getSubnetFromUser(cmd *cobra.Command, r *rosa.Runtime, isSubnetSet bool,
+	cluster *cmv1.Cluster, args MachinePoolArgs) string {
+	var selectSubnet bool
+	var subnet string
+	var err error
+
+	question := "Select subnet for a single AZ machine pool"
+	questionError := "Expected a valid value for subnet for a single AZ machine pool"
+
+	if cluster.Hypershift().Enabled() {
+		question = "Select subnet for a hosted machine pool"
+		questionError = "Expected a valid value for subnet for a hosted machine pool"
+	}
+
+	if !isSubnetSet && interactive.Enabled() {
+		selectSubnet, err = interactive.GetBool(interactive.Input{
+			Question: question,
+			Help:     cmd.Flags().Lookup("subnet").Usage,
+			Default:  false,
+			Required: false,
+		})
+		if err != nil {
+			r.Reporter.Errorf(questionError)
+			os.Exit(1)
+		}
+	} else {
+		subnet = args.Subnet
+	}
+
+	if selectSubnet {
+		subnetOptions, err := getSubnetOptions(r, cluster)
+		if err != nil {
+			r.Reporter.Errorf("%s", err)
+			os.Exit(1)
+		}
+
+		subnetOption, err := interactive.GetOption(interactive.Input{
+			Question: "Subnet ID",
+			Help:     cmd.Flags().Lookup("subnet").Usage,
+			Options:  subnetOptions,
+			Default:  subnetOptions[0],
+			Required: true,
+		})
+		if err != nil {
+			r.Reporter.Errorf("Expected a valid AWS subnet: %s", err)
+			os.Exit(1)
+		}
+		subnet = aws.ParseOption(subnetOption)
+	}
+
+	return subnet
+}
+
+// getSubnetOptions gets one of the cluster subnets and returns a slice of formatted VPC's private subnets.
+func getSubnetOptions(r *rosa.Runtime, cluster *cmv1.Cluster) ([]string, error) {
```

Comment:
NP : Can we get a UT for this please :pray:  I know its implicitly tested in `getSubnetFromUser` but lets be pedantic on this :pray: